### PR TITLE
Fix search bar

### PR DESCRIPTION
**Pull request by pingan1927**
ref #1609 
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [x] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1621)

### CODE REVIEW

File: components/search-bar/index.web.tsx [hunk 1]
```diff
@@ -152,9 +152,17 @@ export default class SearchBar extends React.Component<SearchBarProps, SearchBar
...
+    // 只有支付宝系客户端才完美支持
+    if (navigator.userAgent.indexOf('AlipayClient') > 0) {
+      if (!navigator.userAgent.match(/Android/i) && navigator.userAgent.indexOf('AM') > 0) {
+        // 口碑掌柜iOS 只有5.2以上版本才支持
+      } else {
+        setTimeout(() => {
+          (this.refs as any).searchInput.focus();
+          this.componentDidUpdate();
+        }, 300);
+      }
```
Comment (afc163) [inline components/search-bar/index.web.tsx]: 这个 if else 怪怪的。

Comment (pingan1927) [inline components/search-bar/index.web.tsx]: 哈哈，我也觉得怪怪地，更优雅的写法有吗？

Comment (warmhug) [inline components/search-bar/index.web.tsx]: 测测没问题就好。 只是这种写 alipay agent 特殊代码的做法 能少还是少点好。

Comment (yesmeck) [inline components/search-bar/index.web.tsx]: 用 guard clauses 呀。

Comment (afc163) [inline components/search-bar/index.web.tsx]:
```jsx
const ua = navigator.userAgent;
if (ua.indexOf('AlipayClient') >= 0 && (ua.match(/Android/i) || ua.indexOf('AM') < 0)) {
  // xxx
}
```

**Pull request merged**